File: Objective-C/AdvancedExample/app/Video.m
#import "Video.h"

@implementation Video

- (instancetype)initWithTitle:(NSString *)title
                     assetKey:(NSString *)assetKey
                  networkCode:(NSString *)networkCode
                       apiKey:(NSString *)apiKey {
  self = [super init];
  if (self) {
    _title = [title copy];
    _streamType = StreamTypeLive;
    _assetKey = [assetKey copy];
    _networkCode = [networkCode copy];
    _apiKey = [apiKey copy];
  }
  return self;
}

- (instancetype)initWithTitle:(NSString *)title
              contentSourceId:(NSString *)contentSourceId
                      videoId:(NSString *)videoId
                  networkCode:(NSString *)networkCode
                       apiKey:(NSString *)apiKey {
  self = [super init];
  if (self) {
    _title = [title copy];
    _streamType = StreamTypeVOD;
    _contentSourceID = [contentSourceId copy];
    _videoId = [videoId copy];
    _networkCode = [networkCode copy];
    _apiKey = [apiKey copy];
  }
  return self;
}

@end

File: Swift/AdvancedExample/AdvancedExample/Video.swift
// Copyright 2025 Google LLC. All rights reserved.
//
//
// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
// file except in compliance with the License. You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software distributed under
// the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
// ANY KIND, either express or implied. See the License for the specific language governing
// permissions and limitations under the License.

import Foundation

enum StreamType {
  case live
  case vod
}

class Video {
  let title: String
  let streamType: StreamType
  let assetKey: String?
  let contentSourceID: String?
  let videoID: String?
  let networkCode: String
  let apiKey: String?
  var savedTime: TimeInterval = 0

  init(title: String, assetKey: String, networkCode: String, apiKey: String?) {
    self.title = title
    self.streamType = .live
    self.assetKey = assetKey
    self.contentSourceID = nil
    self.videoID = nil
    self.networkCode = networkCode
    self.apiKey = apiKey
  }

  init(
    title: String, contentSourceID: String, videoID: String, networkCode: String, apiKey: String?
  ) {
    self.title = title
    self.streamType = .vod
    self.assetKey = nil
    self.contentSourceID = contentSourceID
    self.videoID = videoID
    self.networkCode = networkCode
    self.apiKey = apiKey
  }
}
